### PR TITLE
python3Packages: Backport several packages

### PR DESCRIPTION
**Pull request by jsoo1**
Backport changes to several packages that are generally incompatible with each other on NixOS/nixpkgs release-21.05

* aiopg
* anyio
* azure-core
* azure-cosmos
* cryptography
* database
* databases
* fastapi
* importlib-resources
* llvmlite
* numba
* parameterizedtestcase
* python-jose
* smart-open
* starlette
* textacy
* typing
* typing-extensions
* typing-inspect
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/databases/default.nix [hunk 1]
```diff
@@ -13,42 +14,48 @@
...
-  ];
-
-  checkInputs = [
+    aiopg
     aiomysql
     aiosqlite
     asyncpg
     pymysql
+    sqlalchemy
+  ] ++ lib.optionals (pythonOlder "3.7") [
```
Comment (navtej) [inline pkgs/development/python-modules/databases/default.nix:35]: What does exactly `pythonOlder` means?

File: pkgs/development/python-modules/numba/default.nix [hunk 2]
```diff
@@ -12,29 +12,39 @@
 }:
 
 buildPythonPackage rec {
-  version = "0.53.0";
+  version = "0.54.1";
   pname = "numba";
-  # uses f-strings, python 3.9 is not yet supported
-  disabled = pythonOlder "3.6" || pythonAtLeast "3.9";
+  disabled = pythonOlder "3.6" || pythonAtLeast "3.10";
```
Comment (navtej) [inline pkgs/development/python-modules/numba/default.nix:17]: AVA is built on python 3.9. I have not tested3.10. f-strings came way before 3.9 iirc around 3.5/3.6. Do we need py at least at ver 3.10?

**Comment by navtej** [comment]
From the changes in the PR, we need to verify that the following 3 do not break anything.
`azure-core`, 
`azure-cosmos`, 
`typing`

Azure shuffled their repos and organization of azure libs in a consolidation effort, need to cross that with update package everything works.

`typing` is used by Mypy. We might have to pin Mypy to lower version.

I have an integrated test PR which should get merged in a day or two. Post the merge `test-ava-live` should be able to tell us if everything is running fine.

File: pkgs/development/python-modules/databases/default.nix (hunk 1 above)
Comment (jsoo1) [reply pkgs/development/python-modules/databases/default.nix:35]: Hm. Probably! I just cherry-picked this commit from https://github.com/NixOS/nixpkgs which might have a bit of a bug.

File: pkgs/development/python-modules/numba/default.nix (hunk 2 above)
Comment (jsoo1) [reply pkgs/development/python-modules/numba/default.nix:17]: Yeah, this is also cherry-picked from upstream. The reason we do this is so we can drop the patch when we upgrade to the release that has the version we want.

**Comment by jsoo1** [comment]
> I have an integrated test PR which should get merged in a day or two. Post the merge `test-ava-live` should be able to tell us if everything is running fine.

@navtej I ran `test-ava-live` with this revision (nixpkgs pointed to `jsoo1/python3-databases`) and it looks good!

**Pull request merged**